### PR TITLE
docs: Outline the basic workflow.

### PR DESCRIPTION
**Pull request by johnjbarton**
This short narative also helps introduce 'client' and 'context' terminology.

### CODE REVIEW

File: docs/intro/03-how-it-works.md [hunk 1]
```diff
@@ -14,7 +14,28 @@ The server collects the results from all of the captured browsers and presents t
 
 This is only a very brief overview, as the internals of how Karma works aren't entirely necessary when using Karma.
 
-However, if you are interested in learning more, Karma itself originates from a university thesis, which goes into detail about the design
+## Outline of workflow.
+
+Here is roughly how Karma works:
+
+After starting up, Karma loads plugings and the configuration file, then starts its local web server which listened for connections.
```
Comment (SLaks) [inline docs/intro/03-how-it-works.md]: Typos: `plugins`, `listens`

File: docs/intro/03-how-it-works.md [hunk 2]
```diff
@@ -14,7 +14,28 @@ The server collects the results from all of the captured browsers and presents t
...
+Any browser already watching the web address (localhost:9876 by default) will reconnect immediately. Karma also starts reporters
```
Comment (SLaks) [inline docs/intro/03-how-it-works.md]: Can you clarify that "already watching" means (I assume) already have that page loaded from an earlier run?

File: docs/intro/03-how-it-works.md [hunk 3]
```diff
@@ -14,7 +14,28 @@ The server collects the results from all of the captured browsers and presents t
...
+which register for 'browser' events.
```
Comment (SLaks) [inline docs/intro/03-how-it-works.md]: What do you mean?

This sounds like it's saying that it starts those reporters immediately on server startup, whereas I assume you mean that it starts them on (re-)connections.

File: docs/intro/03-how-it-works.md [hunk 4]
```diff
@@ -14,7 +14,28 @@ The server collects the results from all of the captured browsers and presents t
...
+
+Then karma launches zero, one, or more browsers, setting their start page the Karma server URL.
+
+When the browsers connect, Karma serves a 'client.html' page; when this page runs in the browser it connects back to the server via websockets.
+
+Once the server sees the websocket connection, it instructs the client -- over the websocket -- to execute tests.  The client page opens an iframe with a 'context.html' page from the server. The server generates this context.html page using the configuration. This page includes the test framework adapter, the code to be tested, and the test code.
+
+When the browser loads this context page, the onload event handler connect the context page to the client page via post-message. The framework adapter is in charge at this point: it runs the test, reporting errors or success by messaging through the client page.
```
Comment (SLaks) [inline docs/intro/03-how-it-works.md]: Typos: `connects`, `postMessage`.

File: docs/intro/03-how-it-works.md (hunk 2 above)
Comment (johnjbarton) [reply docs/intro/03-how-it-works.md]: Actually I think this is base on the websocket from the client page reconnecting. Changed the text.

File: docs/intro/03-how-it-works.md (hunk 3 above)
Comment (johnjbarton) [reply docs/intro/03-how-it-works.md]: Its part of loading the plugins; clarified.

File: docs/intro/03-how-it-works.md (hunk 4 above)
Comment (johnjbarton) [reply docs/intro/03-how-it-works.md]: done

**Pull request merged**